### PR TITLE
CMakeLists.txt: Add cmake config package export

### PR DESCRIPTION
**Pull request by DasRoteSkelett**
Basically done the same thing as in concurrentqueue to make it easy for cmake based project to pick up the install location / header files.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,10 +2,65 @@ cmake_minimum_required(VERSION 3.9)
...
+                Devel
+)
+
+set(CPACK_PACKAGE_NAME ${PROJECT_NAME})
+set(CPACK_PACKAGE_VENDOR "Cameron Desrochers <cameron@moodycamel.com>")
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "A single-producer, single-consumer lock-free queue for C++.")
+set(CPACK_PACKAGE_VERSION "${PROJECT_VERSION}")
+set(CPACK_PACKAGE_VERSION_MAJOR "${PROJECT_VERSION_MAJOR}")
+set(CPACK_PACKAGE_VERSION_MINOR "${PROJECT_VERSION_MINOR}")
+set(CPACK_PACKAGE_VERSION_PATCH "${PROJECT_VERSION_PATCH}")
```
Comment (cameron314) [inline CMakeLists.txt:62]: Where does this information come from?

Comment (DasRoteSkelett) [reply CMakeLists.txt:62]: It is set in the `project()` command, line 2 of the CMakeLists.txt. (Which right now is 1.0.0, so you might want to bump that). Here are some links to the cmake docs[PROJECT_VERSION](https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION.html#variable:PROJECT_VERSION), [PROJECT_VERSION_MAJOR](https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION_MAJOR.html)

**Comment by cameron314** [comment]
I maintain that I detest the amount of metadata around what is a simple header-only library. However, requests for CMake support seem to be consistent.

**Comment by cameron314** [comment]
Thank you for your contribution.

**Pull request merged**